src/rand.rs

```diff
@@ -229,6 +230,21 @@ mod sysrand_chunk {
     }
 }
 
+#[cfg(target_os = "espidf")]
```

Comment:
This part should be updated to just add `target_os = "espidf"` to the allowlist in the new version of rand.rs, after reviewing that the `getrandom` crate implements this functionality for this operating system correctly. Please do this change in a separate PR.